doc/index.rst

```diff
@@ -184,6 +184,7 @@ PennyLane is **free** and **open source**, released under the Apache License, Ve
    development/guide
    development/plugins
    development/research
+   development/release_notes.md
 
```

Comment:
Do you want this to appear in the "Developers" section in the docs? It does not work in my build, there are only the original three sections...

Reply:
Oh that's weird, it appears in mine :thinking: Does it work if you `make clean && make docs`?

Reply:
Yes, that worked. I also used `make html` accidentally, as in the qml repo. Not sure if that makes a difference.

Reply:
It appears for me, maybe that will indeed be solved by a clean